compiler/noirc_frontend/src/tests.rs

```diff
@@ -4183,3 +4183,21 @@ fn mutable_reference_to_array_element_as_func_arg() {
     "#;
     check_errors(src);
 }
+
+#[test]
+fn object_type_must_be_known_in_method_call() {
+    let src = r#"
+    pub fn foo<let N: u32>() -> [Field; N] {
+        let mut bar = _;
+                      ^ in expressions, `_` can only be used on the left-hand side of an assignment
+                      ~ `_` not allowed here
```

Comment:
Does this really reproduce the original error? I'm not sure if bar's type would be an unbound typevar here or the error type. To guarantee it is unbound you could do something a bit roundabout:
```rs
let array = [];
let bar = array[0]; // this would panic at runtime but we don't run the program
```

Reply:
Nice, thanks! I was looking of a way to get an untyped value but couldn't figure out how. I checked that with `_` the error message is the same, but it just adds noise to the test.